src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -42,19 +42,23 @@ void GcsResourceManager::HandleUpdateResources(
   NodeID node_id = NodeID::FromBinary(request.node_id());
   RAY_LOG(DEBUG) << "Updating resources, node id = " << node_id;
   auto iter = cluster_resources_.find(node_id);
-  auto to_be_updated_resources = request.resources();
+  std::unordered_map<std::string, double> to_be_updated_resources;
+  for (auto &entry : request.resources()) {
+    to_be_updated_resources.emplace(entry.first, entry.second.resource_capacity());
+  }
+
   if (iter != cluster_resources_.end()) {
-    for (auto &entry : to_be_updated_resources) {
+    for (auto &entry : request.resources()) {
       (*iter->second.mutable_items())[entry.first] = entry.second;
     }
+    UpdateResourceCapacity(node_id, to_be_updated_resources);
     auto on_done = [this, node_id, to_be_updated_resources, reply,
                     send_reply_callback](const Status &status) {
       RAY_CHECK_OK(status);
       rpc::NodeResourceChange node_resource_change;
       node_resource_change.set_node_id(node_id.Binary());
       for (auto &it : to_be_updated_resources) {
-        (*node_resource_change.mutable_updated_resources())[it.first] =
-            it.second.resource_capacity();
+        (*node_resource_change.mutable_updated_resources())[it.first] = it.second;
```

Comment:
This is too hard to read. Can you make it like 

```cpp
auto &something = it.first
auto &something2 = it.second
auto &something3 = *node_resource_change.mutable_updated_resources()
something3[something] = something2
```
?

Reply:
Good idea, thanks.